extending/template_system/info_graph.rst

```diff
@@ -0,0 +1,178 @@
+   with the template file (see HTML example linking CSS and JS files).
+ * ``graph``: this object contains all the information from the graph of dependencies. It
+   offers the following API:
+
+   + ``graph.nodes``: list of ``Node`` objects with the information for each Conan package
+     included in the graph (see below API for this ``Node`` object).
+   + ``graph.edges``: list of tuples with all the connections in the dependency graph. First
+     item in the tuple is the consumer ``Node`` and second item the required ``Node``.
+   + ``graph.binary_color(node)``: function that retrieves the Conan default color based on
+     the ``node.binary`` value. 
+
+The ``Node`` objects in the context provide all the required information about each package:
+
+ * ``node.label``: display name for the conanfile.
+ * ``node.short_label``: ``name/version`` parts of the Conan reference.
+ * ``node.package_id``: the package identifier.
+ * ``node.is_build_requires``: 
+ * ``node.binary``: it identifies where the binary comes from (cache, download, build,
+    missing, update). 
+ * ``node.data()``: returns a dictionary that contains data from the recipe, members are ``url``,
+   ``homepage``, ``license``, ``author`` and ``topics``.
+
+
+
+Examples
+--------
+
+This is are two examples of templates Conan is currently using for the basic functionality,
+you can refer to the `Jinja2 documentation <https://palletsprojects.com/p/jinja/>`_ for more
+information about the logic and filters your can use in these templates. 
+
+Let's us know if you have a cool template you want to share with the Conan community.
+
+
+Dot files:
+++++++++++
+
+Default template for the DOT output contains just the node names and the edges:
+
+.. code-block:: text
+
+    digraph {
+        {%- for src, dst in graph.edges %}
+            "{{ src.label }}" -> "{{ dst.label }}"
+        {%- endfor %}
+    }
+
+
+The output will compose a valid ``dot`` file:
+
+.. code-block:: bash
+
+    conan info poco/1.10.0@ --graph=poco.dot
+
+
+.. code-block:: text
+
+    digraph {
+            "poco/1.10.0" -> "openssl/1.1.1g"
+            "virtual" -> "poco/1.10.0"
+    }
+
+
+Use ``dot`` to render the default view of the generated graph:
+
+.. code-block:: bash
+
+    dot -Tpng poco.dot > poco.png
+
+
+.. image:: ../../images/conan-templates-poco.png
+   :width: 240 px
+   :align: center
+
+
+HTML files:
++++++++++++
+
+HTML templates are more complicated than ``dot`` ones, but the HTML can provide a nicer
+view of the graph and easily include JavaScript to create an interactive view of the graph.
+
+In this example we assume you have distributed the following files to your cache folder:
+
+.. code-block:: bash
+
+    <cache>/templates/output/css/vis.min.css
+    <cache>/templates/output/js/vis.min.js
+    <cache>/templates/output/info_graph.html
+
+
+Our template will we the *info_graph.html* file, and it will use the assets from the
```

Comment:
not sure if it's fair to advice usage of **html** extension for the template file. I think something like **tmpl** (or **html.tmpl**) would better illustrate the purpose of the file. it's not an HTML and couldn't be properly rendered via web-browser, thus it shouldn't have an **html** extension IM.

Reply:
I often find myself using other extensions, but then I come back to `.html` for two reasons:
 * Django itself uses `.html` for the HTML templates. Django is a big Python project and whenever I have doubts I usually follow their convention.
 * Even if it is a template, it will contain HTML tags and code and the IDE will render it with colors and help with code completion.

IMO I'll keep using `.html` as the suggestion.